app/forms/additional_code_search_form.rb

```diff
@@ -14,8 +14,10 @@ class AdditionalCodeSearchForm
   attr_writer :page
 
   def validate_code
+    sanitize_code
+
     if code.present?
-      errors.add(:code, :invalid) unless code =~ /\A([A-Z]|[0-9]){4}\z/
```

Comment:
This was checking length and specific characters before. I don't imagine its a huge deal no longer doing the two together but this is a different validation

Reply:
It is just redundant.
Another validation checks if the first char is a valid type. And it works better. It checks if the type is a valid possible_types (and not just generic A-Z).